src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/PrintRequest.kt

```diff
@@ -0,0 +1,133 @@
+package uk.gov.dluhc.printapi.rds.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.UpdateTimestamp
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.database.entity.CertificateFormat
+import uk.gov.dluhc.printapi.database.entity.CertificateLanguage
+import uk.gov.dluhc.printapi.rds.repository.UUIDCharType
+import uk.gov.dluhc.printapi.rds.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.CascadeType
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+import javax.persistence.OneToOne
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class PrintRequest(
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @NotNull
+    @Size(max = 24)
```

Comment:
@Neil-Massey, I think the current implementation does not validate the entity.
We can make it work like this:
<pre>
@<b>field:</b>Size(min=24, max = 24)
</pre>


Do you think it is worth covering the validation in an integration test as well?

Reply:
TBH I'm not sure about the value of these annotations I included as saw they are used in VCA.

Reply:
I think they're worth having (so long as they're in sync with VCA), but I don't think we need the overheard of integration tests for them.

Reply:
@Neil-Massey, @matt-wills-valtech one thing I wanted to point out is the constraint annotations like `@Size` needs to be for the field, for validation to work. Otherwise, it won't work.

For that we need to add `field` to the annotation as below:

`@field:Size`

Reply:
Yes. The point @abdullah-sanver-valtech is making is that because this is a class, and the "fields" here are constructor arguments, simply annotating with `@Size` is annotating the constructor argument, not the backing field itself.
Abdullah is right, we need to use `@field` as a prefix if we expect it to work ....

... which is another discussion in itself .... do we even want/care about these annotations on the entity fields? What is going to actually process them? More importantly, what will happen if it fails said validation ?

(re: the annotation, the kotlin terminology we are talking about here is "annotation use site targets" - https://kotlinlang.org/docs/annotations.html#annotation-use-site-targets )

Reply:
If the annotations are processed (ignoring the specifics of that for a moment), then I would assume it would be best to fail early (hopefully with a meaningful error message), rather than either fail to insert the data, or insert "invalid" data?

Reply:
For my money I would expect any validations to happen long before we go near any persistence layer - request payload validation if it were a REST API or SQS message. How else is data getting into the system? Validate at the point it enters the system?

Reply:
OK - Have activated the annotation for Hibernate validation, and removed all that prevented saving (cannot have size on enums and the spring managed audited properties cannot be null checked)